experimental/ai-algorithms/NeuralNets/Lib.idr

```diff
@@ -0,0 +1,154 @@
+module Lib
+import Prelude.Doubles as db
+
+|||TODO functions not using foldr probably reverses order of elements in a list
+
+export
+stopwords : List String
+stopwords = ["the", "a", "and", "is", "be", "will", "is"] 
+
+export
+nums : String
+nums = "0123456789!@#$%^&*()_+=-:;"
+
+export
+removeStopWords : String -> List String
+removeStopWords line = let tmp = words line
+                       in foldr (\x, acc => if elem x stopwords then acc else x::acc ) [] tmp
+
+             
+||| makeWordPairs creates a list of pairs consisting of words from 
+||| some context n to the left of the focus word and n right to the 
+||| focus word e.g let str = "The prince is the future king"
+||| where n = 2 focus word is The we have (The, prince) (The, is)  
+||| where n = 2 focus word is prince we have (prince, The), (prince, is), (prince, the)....so on 
+
+export
+makeWordPairs : List String -> List (String, String)
+makeWordPairs [] = []
+makeWordPairs (x::y::z::h::xs) = (x,y)::(x,z)::(y,z)::(y,h)::(z,x)::(z,y)::(z,h)::(h,x)::(h,y)::(h,z)::makeWordPairs xs
+makeWordPairs (x::y::z::xs) = (x,y)::(x,z)::(y,z)::(z,x)::(z,y)::makeWordPairs xs
+makeWordPairs (x::y::xs) = (x,y)::makeWordPairs xs
+
+export
+createDictionary : List String ->  List (String, Nat)
+createDictionary xss = func 0 tmp 
+              where
+                tmp = sort . nub $ xss
+                func : Nat -> List String -> List (String, Nat)
+                func _ [] = []
+                func Z (x::xs) = (x, S Z) :: func (S (S Z)) xs
+                func (S k) (x::xs) = (x, S k) :: func (S (S k) ) xs
+
+
+export
+insertAt : List Double -> Double -> Nat -> List Double
+insertAt [] elem pos = []
```

Comment:
Note that `elem` is a standard function, so it might be slightly confusing to use it as variable, it's correct nevertheless.